src/tactic/linear_combination.lean

```diff
@@ -0,0 +1,386 @@
+  sign by subtracting the right side of the equation from the left side.  In
+  other words, given lhs = rhs, this tactic returns lhs - rhs = 0.
+
+* Input:
+  * `heq` : an expr, which should be an equality with some type α on each side,
+      where `add_group α` is true
+
+* Output: tactic expr that is lhs - rhs = 0, where lhs and rhs are the left and
+  right sides of heq respectively
+-/
+meta def move_to_left_side (heq : expr) : tactic expr :=
+do
+  tactic.mk_app ``left_minus_right [heq]
+  <|> tactic.fail ("The type of the left and right sides of each equality " ++
+    "must fulfill the 'add_group' condition in order to match the linear " ++
+    "combination to the target.")
+
+/--
+Moves all the terms in the target to the left side of the equals sign by
+  subtracting the right side of the equation from the left side.  In other
+  words, given a target of lhs = rhs, this tactic returns lhs - rhs = 0.
+Note: The target must be an equality when this tactic is called, and the
+  equality must have some type α on each side, where `add_group α` is true.
+
+* Input: N/A
+
+* Output: tactic unit
+-/
+meta def move_target_to_left_side : tactic unit :=
+do
+  -- Move all the terms in the target equality to the left side
+  target ← tactic.target,
+  (targ_lhs, targ_rhs) ← tactic.match_eq target,
+  target_left_eq ← tactic.to_expr ``(%%targ_lhs - %%targ_rhs = 0),
+  do
+    { can_replace_proof ← tactic.mk_app ``all_on_left_equiv [targ_lhs, targ_rhs],
+      tactic.replace_target target_left_eq can_replace_proof }
+  <|> tactic.fail ("The type of the left and right sides of the goal " ++
+    "must fulfill the 'add_group' condition in order to match the linear " ++
+    "combination to the target.")
+
+
+
+/-! ### Part 3: Matching the Linear Combination to the Target -/
+
+
+/--
+This tactic changes the goal to be that the lefthand side of the target is
+  equal to the lefthand side of the given expression.  For example,
+  if `hsum_on_left` is 5*x - 5*y = 0, and the target is -5*y + 5*x = 0, this
+  tactic will change the target to be -5*y + 5*x = 5*x - 5*y.
+
+This tactic only should be used when the target is an equality whose right side
+  is 0.
+
+* Input:
+  * `hsum_on_left` : expr, which should be an equality whose right side is 0
+
+* Output: tactic unit
+-/
+meta def set_goal_to_hleft_eq_tleft (hsum_on_left : expr) : tactic unit :=
+do
+  { change_goal ← tactic.to_expr ``(replace_eq_expr %%hsum_on_left),
+    tactic.apply change_goal,
+    pure () }
+  <|> tactic.fail ("The type of the left and right sides of each equality " ++
+    "must fulfill the 'has_zero' condition.")
+
+/--
+This tactic attempts to prove the goal by normalizing the target if the
+`normalize` field of the given configuration is true.
+
+* Input:
+  * `config` : a linear_combination_config, which determines the tactic used
+      for normalization if normalization is done
+
+* Outout: tactic unit
```

Comment:
```suggestion
* Output: tactic unit
```
Although, `tactic unit` is essentially `void` (and it's clear from the declaration) so this isn't really necessary.

Reply:
Good catch!  Fixed!